### ISSUE TITLE
Listener cleanup

### ISSUE THREAD

**Issue by mohadib** [brunob/leaflet.fullscreen#70]
Hey,

 I had an issue where if I called ```map.remove()``` and then reinitialized the map again, then went full screen I would get an error.

```
Uncaught TypeError: Cannot read property '_leaflet_pos' of undefined
    at getPosition (leaflet-src.js:2565)
    at NewClass._getMapPanePos (leaflet-src.js:4110)
    at NewClass._rawPanBy (leaflet-src.js:3905)
    at NewClass.invalidateSize (leaflet-src.js:3293)
    at NewClass._handleFullscreenChange (Control.FullScreen.js:114)
    at HTMLDocument.handler (leaflet-src.js:2108)
```

I was able to fix this by cleaning up the listeners on the fullscreen button and removing the control. Hopefully this will help someone else.

```	
onRemove: function () {
		L.DomEvent
			.removeListener(this.link, 'click', L.DomEvent.stopPropagation)
			.removeListener(this.link, 'click', L.DomEvent.preventDefault)
			.removeListener(this.link, 'click', this.toggleFullScreen, this);

		L.DomEvent
			.removeListener(this.container, fullScreenApi.fullScreenEventName, L.DomEvent.stopPropagation)
			.removeListener(this.container, fullScreenApi.fullScreenEventName, L.DomEvent.preventDefault)
			.removeListener(this.container, fullScreenApi.fullScreenEventName, this._handleFullscreenChange, this);

		L.DomEvent
			.removeListener(document, fullScreenApi.fullScreenEventName, L.DomEvent.stopPropagation)
			.removeListener(document, fullScreenApi.fullScreenEventName, L.DomEvent.preventDefault)
			.removeListener(document, fullScreenApi.fullScreenEventName, this._handleFullscreenChange, this);
	},
```

**Comment by brunob** [comment]
Thx for the report, it may be related to #69 ; could you test this patch and tell me if it fix your bug ?

**Comment by Raruto** [comment]
Hi bruno,
I also experienced a similar problem when removing the control from a map (and after that when another element goes "fullscreen" in the same page).

(While waiting for a future patch), here's how to overcome this little bug without editing the plugin code:
```js
  // Save a reference of default "onRemove" function (for later use)
  var fullscreenProto = L.Control.FullScreen.prototype;
  var onRemoveFullScreenProto = fullscreenProto.onRemove;

 // Customize default "L.Control.FullScreen" behaviour.
  L.Control.FullScreen.include({
    onRemove: function(map) {

      // Call your default "onRemove" function (if there will be any..)

      if (onRemoveFullScreenProto) {
        onRemoveFullScreenProto.call(this, map);
        return;
      }

      // Proposal fix for: #69 and #70

      L.DomEvent
        .removeListener(this.link, 'click', L.DomEvent.stopPropagation)
        .removeListener(this.link, 'click', L.DomEvent.preventDefault)
        .removeListener(this.link, 'click', this.toggleFullScreen, this);
      L.DomEvent
        .removeListener(this._container, fullScreenApi.fullScreenEventName, L.DomEvent.stopPropagation)
        .removeListener(this._container, fullScreenApi.fullScreenEventName, L.DomEvent.preventDefault)
        .removeListener(this._container, fullScreenApi.fullScreenEventName, this._handleFullscreenChange, this);
      L.DomEvent
        .removeListener(document, fullScreenApi.fullScreenEventName, L.DomEvent.stopPropagation)
        .removeListener(document, fullScreenApi.fullScreenEventName, L.DomEvent.preventDefault)
        .removeListener(document, fullScreenApi.fullScreenEventName, this._handleFullscreenChange, this);
    },
  });
```
_Have a nice day,
Raruto_

**Comment by brunob** [comment]
Fixed in 1.6.0, thx for the patch :)
